local/network.go

```diff
@@ -452,6 +454,9 @@ func (ln *localNetwork) addNode(nodeConfig node.Config) (node.Node, error) {
 	}
 	addNetworkFlags(ln.log, ln.flags, nodeConfig.Flags)
 
+	// TODO: remove this when plugin-dir is available
+	delete(nodeConfig.Flags, PluginDirKey)
```

Comment:
What's the purpose of this line?

Reply:
to avoid giving the plugin-dir param to avalanchego. current avalanchego does not accepts the flag, and makes CI to fail

Reply:
I understand with these, plugin-dir can not be used in ANR. But I think this does not matter and can make that avalanchego branch to pass CI (what was asked was not to use build-dir which is completely avoided, while doing that, also this PR partially prepares for using plugin-dir flag).

Reply:
Is this correct @danlaine ?

Reply:
This branch should assume that `build-dir` is no longer a flag, and that the user may pass a flag `plugin-dir`.  So I'm not sure that we want to delete this flag here

Reply:
Ok I did not want to change certain things because by default ANR creates plugin-dir flags without being requested by the user, so making ANR CI to fail. Ok will revert this is the flag in needed to be used on ANR PR, and comment other parts.